OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -124,6 +126,13 @@ public void Killed(Actor self, AttackInfo e)
 				attackerStats.KillsCost += cost;
 				defenderStats.DeathsCost += cost;
 			}
+
+			var experienceInfo = self.Info.TraitInfo<GivesExperienceInfo>();
+			if (experienceInfo != null && experienceInfo.GivesPlayerExperience)
+		    {
+				var exp = self.Info.TraitInfo<ValuedInfo>().Cost;
```

Comment:
Line https://github.com/OpenRA/OpenRA/pull/9536/files#diff-d10fad95a9ee072ed5c227a38d1e35ffR125 already does the valued info lookup so you could move the local variable up and re-use that as well for bonus points.